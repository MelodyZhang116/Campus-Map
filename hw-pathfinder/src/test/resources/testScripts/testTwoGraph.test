# test two graphs so that they affect each other

CreateGraph complexGraph

AddNode complexGraph A
AddNode complexGraph B
AddNode complexGraph C
AddNode complexGraph D
AddNode complexGraph E

AddEdge complexGraph A B 1.0
AddEdge complexGraph B E 100.0
AddEdge complexGraph C D 50.0
AddEdge complexGraph D E 1.0
AddEdge complexGraph A C 20.0

FindPath complexGraph A E

CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B
AddNode simpleGraph C
AddNode simpleGraph D

AddEdge simpleGraph A B 10.0
AddEdge simpleGraph A C 1.0
AddEdge simpleGraph C D 1.0
AddEdge simpleGraph D B 1.0

FindPath simpleGraph A B

FindPath complexGraph A E
