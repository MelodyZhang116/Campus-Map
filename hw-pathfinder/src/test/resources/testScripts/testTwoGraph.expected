# test two graphs so that they affect each other

created graph complexGraph

added node A to complexGraph
added node B to complexGraph
added node C to complexGraph
added node D to complexGraph
added node E to complexGraph

added edge 1.000 from A to B in complexGraph
added edge 100.000 from B to E in complexGraph
added edge 50.000 from C to D in complexGraph
added edge 1.000 from D to E in complexGraph
added edge 20.000 from A to C in complexGraph

path from A to E:
A to C with weight 20.000
C to D with weight 50.000
D to E with weight 1.000
total cost: 71.000

created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph
added node C to simpleGraph
added node D to simpleGraph

added edge 10.000 from A to B in simpleGraph
added edge 1.000 from A to C in simpleGraph
added edge 1.000 from C to D in simpleGraph
added edge 1.000 from D to B in simpleGraph

path from A to B:
A to C with weight 1.000
C to D with weight 1.000
D to B with weight 1.000
total cost: 3.000

path from A to E:
A to C with weight 20.000
C to D with weight 50.000
D to E with weight 1.000
total cost: 71.000